magic/oracle.py

```diff
@@ -35,17 +35,25 @@ def load_cards(names: Iterable[str] = None, where: Optional[str] = None) -> List
     else:
         setnames = set()
     if setnames:
-        names_clause = 'c.name IN ({names})'.format(names=', '.join(sqlescape(name) for name in setnames))
+        names_clause = 'name IN ({names})'.format(names=', '.join(sqlescape(name) for name in setnames))
     else:
         names_clause = '(1 = 1)'
     if where is None:
         where = '(1 = 1)'
     sql = multiverse.cached_base_query('({where} AND {names})'.format(where=where, names=names_clause))
-    rs = db().select(sql)
-    if setnames and len(setnames) != len(rs):
-        missing = setnames.symmetric_difference([r['name'] for r in rs])
-        raise TooFewItemsException('Expected `{namelen}` and got `{rslen}` with `{names}`.  missing=`{missing}`'.format(namelen=len(setnames), rslen=len(rs), names=setnames, missing=missing))
-    return [Card(r) for r in rs]
+    rows = db().select(sql)
+    if setnames and len(setnames) != len(rows):
+        missing = setnames.symmetric_difference([row['name'] for row in rows])
+        raise TooFewItemsException('Expected `{namelen}` and got `{rowslen}` with `{names}`.  missing=`{missing}`'.format(namelen=len(setnames), rowslen=len(rows), names=setnames, missing=missing))
+    ret = []
+    for row in rows:
+        c = Card(row)
+        if c.colors:
```

Comment:
This property does not exist so crashes here.

Reply:
Or maybe you brought it into existence but we crash before the db migration runs?

Reply:
Okay yeah. I made some changes, then decided to do things a different way and undid them. But I failed to undo them everywhere. I can fix this.

Reply:
Is there a way to apply migrations to the cards database? the magic/ directory doesn't have a sql/ subdir.

Reply:
Cards db is entirely dropped and recreated when it changes. You can make the changes to database/multiverse/card and then bump `SCHEMA_VERSION = 106` in database.py if you want to change the schema